lib/doorkeeper/openid_connect.rb

```diff
@@ -69,12 +69,14 @@ def self.signing_key_normalized
       Doorkeeper::GrantFlow.register(
         :id_token,
         response_type_matches: 'id_token',
+        response_mode_matches: %w[fragment form_post],
         response_type_strategy: Doorkeeper::OpenidConnect::IdToken,
       )
 
       Doorkeeper::GrantFlow.register(
         'id_token token',
         response_type_matches: 'id_token token',
+        response_mode_matches: %w[fragment form_post],
```

Comment:
**question:** Do we need to add `query` as well? It's currently listed at https://github.com/doorkeeper-gem/doorkeeper-openid_connect/blob/805b7ca9c42a011e357b2d6daac88c45e7f2172a/app/controllers/doorkeeper/openid_connect/discovery_controller.rb#L41

We should also add `form_post` in the discovery response there, maybe we can do it dynamically with something like this:

```diff
diff --git i/app/controllers/doorkeeper/openid_connect/discovery_controller.rb w/app/controllers/doorkeeper/openid_connect/discovery_controller.rb
index c6fa4c3..09e139f 100644
--- i/app/controllers/doorkeeper/openid_connect/discovery_controller.rb
+++ w/app/controllers/doorkeeper/openid_connect/discovery_controller.rb
@@ -38,7 +38,7 @@ module Doorkeeper
 
           # TODO: support id_token response type
           response_types_supported: doorkeeper.authorization_response_types,
-          response_modes_supported: %w[query fragment],
+          response_modes_supported: response_modes_supported(doorkeeper),
           grant_types_supported: grant_types_supported(doorkeeper),
 
           # TODO: look into doorkeeper-jwt_assertion for these
@@ -70,6 +70,10 @@ module Doorkeeper
         }.compact
       end
 
+      def response_modes_supported(doorkeeper)
+        doorkeeper.authorization_response_flows.flat_map(&:response_mode_matches).uniq
+      end
+
       def grant_types_supported(doorkeeper)
         grant_types_supported = doorkeeper.grant_flows.dup
         grant_types_supported << 'refresh_token' if doorkeeper.refresh_token_enabled?
```

Reply:
@toupeira 

> question: Do we need to add query as well? It's currently listed at

The `id_token token` response_type indicates that the implicit grant flow is used. So we _must not_ add `query` response mode to implicit grant flow.

That is written in openid spec as following. https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Combinations
> id_token token
...The **default Response Mode** for this Response Type **is the fragment encoding** and **the query encoding MUST NOT be used**...

(I also mentioned it in [this pull](https://github.com/doorkeeper-gem/doorkeeper/pull/1438) )
> Implicit Flow:
"query" is not allow because with this response_mode, sensitive data (access token, id token (oidc)) can be seen by user/browser history/server log)

Reply:
I added response_modes_supported and test at here https://github.com/doorkeeper-gem/doorkeeper-openid_connect/pull/138/commits/5b5b4695229b1d878c50b00a1b510fcae69ce262

Reply:
> The id_token token response_type indicates that the implicit grant flow is used. So we must not add query response mode to implicit grant flow.

Thanks for checking! :+1: